main.go

```diff
@@ -194,23 +200,57 @@ func main() {
 
 	srv := &http.Server{
 		Handler: router,
-		Addr:    listen,
 		// Good practice: enforce timeouts for servers you create!
 		WriteTimeout: time.Duration(cfg.Cfg.WriteTimeout) * time.Second,
 		ReadTimeout:  time.Duration(cfg.Cfg.ReadTimeout) * time.Second,
 		IdleTimeout:  time.Duration(cfg.Cfg.IdleTimeout) * time.Second,
 		ErrorLog:     log.New(&fwdToZapWriter{fastlog}, "", 0),
 	}
 
+	lis, cleanupFn, err := listen(cfg.Cfg)
+	if err != nil {
+		logger.Fatal(err)
+	}
+	defer cleanupFn()
+
 	if tls {
 		srv.TLSConfig = cfg.TLSConfig(cfg.Cfg.TLS.Profile)
-		logger.Fatal(srv.ListenAndServeTLS(cfg.Cfg.TLS.Cert, cfg.Cfg.TLS.Key))
+		logger.Fatal(srv.ServeTLS(lis, cfg.Cfg.TLS.Cert, cfg.Cfg.TLS.Key))
 	} else {
-		logger.Fatal(srv.ListenAndServe())
+		logger.Fatal(srv.Serve(lis))
 	}
 
 }
 
+func listen(config *cfg.Config) (lis net.Listener, cleanupFn func(), err error) {
```

Comment:
for consistency, instead of passing `config *cfg.Config` please just use `cfg.Cfg.___` and leave the signature as `listen()`.  Config isn't meant to be passed around, just plucked from the ether as a global.

Reply:
Updated the listen function to use the global configuration object